### PR TITLE
develop apha parser

### PR DESCRIPTION
**Pull request by chaoyinYang**

### CODE REVIEW

File: apha/parser.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    result.mime     = 'HTML';
+    result.title_id = match[2];
+    result.unitid   = `${match[1]}/${match[2]}`;
+
+  } else if ((match = /^\/doi\/epub\/([0-9.]+)\/([0-9]+)$/i.exec(path)) !== null) {
+    // https://pharmacylibrary.com/doi/epub/10.21019/9781582122168
+    // https://pharmacylibrary.com/doi/epub/10.21019/9781582122861
+    result.rtype    = 'BOOK';
+    result.mime     = 'EPUB';
+    result.doi      = `${match[1]}/${match[2]}`;
```
Comment (felixleo22) [inline apha/parser.js:64]: This is not a valid DOI, I have tried it on https://www.doi.org
it looks more like a DOI prefix and an ISBN. match[2] as ISBN.
I don't get the impression that this platform offers the full DOI in its URLs. Unlucky.

File: apha/parser.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    result.unitid   = `${match[1]}/${match[2]}`;
+  } else if ((match = /^\/doi\/epdf\/([0-9.]+)\/([0-9]+)$/i.exec(path)) !== null) {
+    // https://pharmacylibrary.com/doi/epdf/10.21019/9781582121444
+    result.rtype    = 'BOOK';
+    result.mime     = 'PDF';
+    result.doi      = `${match[1]}/${match[2]}`;
```
Comment (felixleo22) [inline apha/parser.js:70]: DOI is not valid : match[2] as ISBN.

**Review by felixleo22 — CHANGES_REQUESTED** [review]

File: apha/parser.js (hunk 1 above)
Comment (chaoyinYang) [reply apha/parser.js]: Hi @felixleo22 
Sure. I removed the doi. Please check again.

File: apha/parser.js (hunk 2 above)
Comment (chaoyinYang) [reply apha/parser.js]: Sure. I removed doi.

**Comment by felixleo22** [comment]
My bad, now these DOI are recognised by https://www.doi.org. I don't know why it wasn't valid at the time, perhaps I made a typing error. You can apply the old commit and I will merge this PR.

**Review by felixleo22 — APPROVED** [review]

**Comment by felixleo22** [comment]
I'm going to merge this PR, but I'm going to add the DOI to the parser, so don't forget to pull the repository to get my changes.

**Pull request merged**